### ISSUE TITLE
Camunda Cloud disconnected after a few hours

### ISSUE THREAD

**Issue by pabloromanelli** [zeebe-io/zeebe-script-worker#25]
Hi, about an hour and a half after starting this worker to Camunda Cloud, the connection gets reseted:
```
13:28:01.234 [grpc-default-executor-0] INFO  io.zeebe.client.job.poller - Activated 1 jobs for worker script-worker and job type script
13:28:01.320 [pool-3-thread-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(org.camunda.feel.spi.JavaValueMapper@13e7eca1)), function-provider: org.camunda.feel.interpreter.FunctionProvider$EmptyFunctionProvider$@1f82136c]
15:03:10.551 [grpc-default-executor-1] WARN  io.zeebe.client.job.poller - Failed to activated jobs for worker script-worker and job type script
io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:533) ~[grpc-api-1.26.0.jar:1.26.0]
...
Caused by: java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[?:?]
...
```

After that, it never reconnects again. It continuously fails with:
```
15:03:12.599 [grpc-default-executor-0] WARN  io.zeebe.client.job.poller - Failed to activated jobs for worker script-worker and job type script
io.grpc.StatusRuntimeException: UNAUTHENTICATED: HTTP status code 401
invalid content-type: text/html
headers: Metadata(:status=401,server=nginx/1.17.7,date=Tue, 10 Mar 2020 18:03:12 GMT,content-type=text/html,content-length=179,strict-transport-security=max-age=15724800; includeSubDomains)
DATA-----------------------------
<html>
<head><title>401 Authorization Required</title></head>
<body>
<center><h1>401 Authorization Required</h1></center>
<hr><center>nginx/1.17.7</center>
</body>
</html>

	at io.grpc.Status.asRuntimeException(Status.java:533) ~[grpc-api-1.26.0.jar:1.26.0]
...
```

It uses `zeebe-client-java` version `0.22.1` and its configured with the following env variables:
- `zeebe.client.broker.contactPoint` = `XXXXXXX.zeebe.camunda.io:443`
- `ZEEBE_INSECURE_CONNECTION`= `false`
- `ZEEBE_CLIENT_ID` = `XXXXXX`
- `ZEEBE_CLIENT_SECRET` = `XXXXXXX`

Camunda Cloud cluster info:
- Generation: Zeebe 0.22.1
- Type: Development
- Status: Healthy

If I manually stop and start the worker again, it reconnects successfully and works as expected.

**Comment by saig0** [comment]
Thank you for reporting! This seems to be a failure in `zeebe-client-java` and not specific to this worker.

**Comment by pabloromanelli** [comment]
Oh sorry, I found this issue on the zeebe repo: https://github.com/zeebe-io/zeebe/issues/3768
Apparently the fix will be included in the 0.23 client.
